agoradatatools/etl/load.py

```diff
@@ -52,12 +65,15 @@ def remove_non_values(d: dict) -> dict:
             if len(nested_dict.keys()) > 0:
                 cleaned_dict[key] = nested_dict
         # case 2: list
-        if isinstance(value, list):
-            for elem in value:  # value is a list
+        elif isinstance(
+            value, list
+        ):  # was missing elif before - was just if and was breaking the recursion
+            for i, elem in enumerate(value):  # value is a list
                 if isinstance(elem, dict):
-                    nested_dict = remove_non_values(elem)
-                else:
-                    cleaned_dict[key] = value
+                    value[i] = remove_non_values(elem)
```

Comment:
similar to the description of the nested dictionary case, dictionaries within lists were also not being cleaned. My proposed solution here loops through the list, runs the `remove_non_values` function on dictionaries, and then removes empty dictionaries from the list that would result from a case such as `"l": ["m", "n", {"o": None}]`

Reply:
Do the empty dictionaries matter? I wasn't sure, but I didn't think that it would hurt to remove them

Reply:
Yes, per our discussion, my recommendation is to :

1. Run the code on the test configuration and compare older version with newer version
2. Discuss with Jess about this change if there are differences to see if it may affect downstream usage of data.

Reply:
Confirmed that the output does not change